addons/hr_payroll/wizard/hr_payroll_payslips_by_employees.py

```diff
@@ -3,17 +3,22 @@
 
 from odoo import api, fields, models, _
 from odoo.exceptions import UserError
+from datetime import datetime
 
 
 class HrPayslipEmployees(models.TransientModel):
     _name = 'hr.payslip.employees'
     _description = 'Generate payslips for all selected employees'
 
-    employee_ids = fields.Many2many('hr.employee', 'hr_employee_group_rel', 'payslip_id', 'employee_id', 'Employees',
-                                    domain=lambda self: self._get_available_contracts_domain())
-
     def _get_available_contracts_domain(self):
-        return [('contract_ids.state', 'in', ('open', 'pending'))]
+        return [('contract_ids.state', 'in', ('open', 'pending')), ('company_id', '=', self.env.user.company_id.id)]
```

Comment:
once again, shouldn't the per-company filtering be done by the default record rules? why hardcode the rules, preventing any customisation by the admin?

Reply:
The ir.rule comes in 12.2 sadly.